### ISSUE TITLE
aws_cloudwatch_log_group fails to create resources with name_prefix

### ISSUE THREAD

**Issue by sfrode** [terraform-providers/terraform-provider-aws#1752]
Creating an "_aws_cloudwatch_log_group_" resource with "_name_prefix_"  will fail:

```
* module.vpc.aws_cloudwatch_log_group.flowlogs_log_group: 1 error(s) occurred:
* aws_cloudwatch_log_group.flowlogs_log_group: InvalidParameter: 1 validation error(s) found.
- minimum field size of 1, PutRetentionPolicyInput.LogGroupName.
```
This is most likely because "_aws_cloudwatch_log_group_" is using _d.Get("name")_ throughout even though "_name_" is an optional variable and not defined in the data structure when "_name_prefix_" is used.

### Terraform Version
0.10.6 with AWS provider 0.1.4 (and master)

### Affected Resource(s)
- aws_cloudwatch_log_group

### Terraform Configuration Files
```hcl
resource "aws_cloudwatch_log_group" "flowlogs_log_group" {
  retention_in_days = 7
  name_prefix       = "FlowLogs-VPC-${var.name}-"
}
```
### Debug Output
Relevant parts:
https://gist.github.com/sfrode/ecd3c9054a37f5c2ca9af7a958ea5d20

### Expected Behavior
No Errors

### Actual Behavior
Failed to set proper retention on resource due to a failure in the resourceAwsCloudWatchLogGroupUpdate() call:

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
None

### References
None as far as I can tell.

**Comment by Burmuley** [comment]
Have exactly the same behavior. Please fix asap.

**Comment by Ninir** [comment]
Closed by #2190
